src/core/atomicdex/pages/qt.settings.page.cpp

```diff
@@ -606,12 +585,84 @@ namespace atomic_dex
             }
         };
 
-        functor_remove(std::move(wallet_cfg_path));
+        if (fs::exists(wallet_cfg_path))
+        {
+            nlohmann::json wallet_config_json_data;
+            nlohmann::json active_list = nlohmann::json::array();
+            QFile          fs;
+            fs.setFileName(std_path_to_qstring(wallet_cfg_path));
+            fs.open(QIODevice::ReadOnly | QIODevice::Text);
+
+            //! Read Contents
+            wallet_config_json_data = nlohmann::json::parse(QString(fs.readAll()).toStdString());
+            fs.close();
+
+            //! Get the active coins
+            for (auto&& [key, value]: wallet_config_json_data.items())
+            {
+                if (value["active"]) { active_list.push_back(key); }
+            }
+
+            // remove old coins file
+            functor_remove(std::move(wallet_cfg_path));
+
+            //! Copy default coins file
+            const auto  cfg_path = ag::core::assets_real_path() / "config";
+            std::string filename = std::string(atomic_dex::get_raw_version()) + "-coins.json";
+            fs::copy(cfg_path / filename, wallet_cfg_path);
+
+            //! Open coins file
+            fs.setFileName(std_path_to_qstring(wallet_cfg_path));
+            fs.open(QIODevice::ReadOnly | QIODevice::Text);
+
+            //! Read Contents
+            wallet_config_json_data = nlohmann::json::parse(QString(fs.readAll()).toStdString());
+            fs.close();
+
+            //! set active coins again
+            for (auto&& [key, value]: wallet_config_json_data.items())
+            {
+                if (std::find(active_list.begin(), active_list.end(), key) != active_list.end())
+                {
+                    value["active"] = true;
+                }
+            }
```

Comment:
Here by using the `unordered_set<string>` approach you will reduce drastically your complexity <https://www.bigocheatsheet.com/>

you have now an O(n)² complexity.

you will be at O(n) complexity afterwards.